ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -210,13 +223,11 @@ def _reset_env(self, env: BaseUnityEnvironment):
             A Data structure corresponding to the initial reset state of the
             environment.
         """
-        if self.meta_curriculum is not None:
-            return env.reset(
-                train_mode=self.fast_simulation,
-                config=self.meta_curriculum.get_config(),
-            )
-        else:
-            return env.reset(train_mode=self.fast_simulation)
+        sampled_reset_param = self.sampler_manager.sample_all()
+        new_meta_curriculum_config = (self.meta_curriculum.get_config() 
+                                    if self.meta_curriculum else {})
+        sampled_reset_param.update(new_meta_curriculum_config)
```

Comment:
Does this mean meta_curriculum would override sampler_manager if provided? I'd just make sure this is clear in the documentation, or say that you have to use one or the other.

Reply:
Only overrides if they conflict on a key. One of them has to take priority, not sure which one it should be though.

Reply:
As of now, meta_curriculum takes priority

Reply:
meta_curriculum does take priority over generalization training as this made most sense.